lib/generator._js

```diff
@@ -296,55 +296,15 @@ ScriptGenerator.prototype.generateWapDeploymentScript = function (_) {
-        var webProjFiles = files.filter(function(file){
-            return file.endsWith(".xproj");
-        });
-        if (webProjFiles.length != 1) {
-            throw new Error('Expecting one xproj files but found ' + webProjFiles.length + ' under' + this.absoluteProjectPath);
-        }
-        this.projectPath = path.relative(this.repositoryRoot, path.join(this.absoluteProjectPath, webProjFiles[0]));
-    }
-
-    var msbuildArguments;
-    var dotnetpublishArguments;
-
-    if (this.scriptType == ScriptType.batch) {
-      msbuildArguments = '"%DEPLOYMENT_SOURCE%\\' + this.solutionPath + '" /nologo /verbosity:m /p:AutoParameterizationWebConfigConnectionStrings=false;Configuration=Release;UseSharedCompilation=false %SCM_BUILD_ARGS%';
-      dotnetpublishArguments = '"%DEPLOYMENT_SOURCE%\\' + this.projectPath + '" /nologo /verbosity:m /t:GatherAllFilesToPublish /p:AutoParameterizationWebConfigConnectionStrings=false;Configuration=Release;UseSharedCompilation=false;PublishOutputPathNoTrailingSlash="%DEPLOYMENT_TEMP%" %SCM_BUILD_ARGS%';
-    } else {
-      msbuildArguments = '"$DEPLOYMENT_SOURCE\\' + this.solutionPath + '" /nologo /verbosity:m /p:AutoParameterizationWebConfigConnectionStrings=false`;Configuration=Release`;UseSharedCompilation=false` $SCM_BUILD_ARGS';
-      dotnetpublishArguments = '"$DEPLOYMENT_SOURCE\\' + this.projectPath + '" /nologo /verbosity:m /t:GatherAllFilesToPublish /p:AutoParameterizationWebConfigConnectionStrings=false`;Configuration=Release`;UseSharedCompilation=false`;PublishOutputPathNoTrailingSlash="$DEPLOYMENT_TEMP" $SCM_BUILD_ARGS';
-    }
+  // this.solutionPath and this.projectPath are both relative
+  var options = {};
+  options.RestoreArguments = ((this.solutionPath) ? this.solutionPath:this.projectPath);
+  options.DotnetpublishArguments = this.projectPath;
 
-    var options = {
-      msbuildArguments: msbuildArguments,
-      dotnetpublishArguments: dotnetpublishArguments,
-      restore: nugetRestore,
-      solutionPath: this.solutionPath
-    };
+  this.generateAspNetCoreScript('aspnet.core.template', options, _);
 
-    this.generateAspNetCoreScript('aspnet.core.msbuild.template', options, _);
-  } else {
-    // use dotnet command, not msbuild if NO solutionfile is provided
-    // or if there's a solution file, but preview3 project
-    var nugetOrDotnetRestore = this.projectPath.endsWith(".csproj") ? 'dotnet restore': nugetRestore;
-    if (this.solutionPath) {
-        nugetOrDotnetRestore += ' "' + this.solutionPath + '"';
-    } else {
-        nugetOrDotnetRestore += ' "' + this.projectPath + '"';
-        // current directory does not have .sln or .csproj/.xproj, nuget restore will throw an error
-    }
-    this.generateAspNetCoreScript('aspnet.core.template', {aspNetCoreProject: this.projectPath, restore:nugetOrDotnetRestore}, _);
-  }
 };
 
 ScriptGenerator.prototype.generateDnxConsoleAppDeploymentScript = function (_) {
```

Comment:
no, I am removing the function + template since the logic was already gone in this [commit](https://github.com/projectkudu/KuduScript/commit/bec1170f47c61a6529e6b41d0b6a98832cfc9b59#diff-6ce8a1e94766242af3343f6b85f4f9d4L132)